src/renderers/webgl/WebGLUniforms.js

```diff
@@ -569,6 +568,13 @@ function setValueT1( gl, v, textures ) {
 
 	}
 
+	if ( this.type === gl.SAMPLER_2D_SHADOW && emptyShadowTexture === null ) {
```

Comment:
Would it also fix the tree shaking issue if we do the following?
```js
let emptyTexture2D;

if ( this.type === gl.SAMPLER_2D_SHADOW ) {

    emptyShadowTexture.compareFunction = LessEqualCompare;
    emptyTexture2D = emptyShadowTexture;

} else {

    emptyTexture2D = emptyTexture;

}
```
I suspect tree-shaking breaks since the code accesses `compareFunction` right away. If we just move this bit to `setValueT1()`, we don't need an additional `if` check.

Reply:
Yes, anything were we don't do a mutation in the top-level scope.